workflows/mcmicro.nf

```diff
@@ -74,19 +74,20 @@ workflow MCMICRO {
     ch_versions = ch_versions.mix(ASHLAR.out.versions)
 
     // // Run Background Correction
-    // BACKSUB(ASHLAR.out.tif, ch_markers)
-    //BACKSUB(ASHLAR.out.tif, [[id: "backsub"], params.marker_sheet])
-    /*
     if (params.backsub) {
-        BACKSUB(ASHLAR.out.tif, [[id:"$ASHLAR.out.tif[0]['id']"], params.marker_sheet])
+        ch_backsub_markers = ch_markersheet
+            .map { ['channel_number,cycle_number,marker_name,exposure,background,remove',
+                it.collect{ channel_number, cycle_number, marker_name, _1, _2, _3, exposure, background, remove ->
+                    channel_number + "," + cycle_number + "," + marker_name + "," + exposure + "," + background + "," + remove}] }
+            .flatten()
+            .map { it.replace('[]', '') }
+            .collectFile(name: 'markers_backsub.csv', sort: false, newLine: true, storeDir: "${workDir}")
+        BACKSUB(ASHLAR.out.tif, [[id:"$ASHLAR.out.tif[0]['id']"], "${workDir}/markers_backsub.csv"])
```

Comment:
As mentioned above, instead of an explicit path for the csv, use `ch_backsub_markers`. You'll need to use `combine` to join the collectFile-returned single-value channel with each entry in the meta-map channel for the second arg to BACKSUB.

Reply:
Got it.  Thank you for the clear instructions!  This is much better.